week-1-ts-ecosystem/w01e04-light-me-up/src/light.ts

```diff
@@ -0,0 +1,15 @@
+import { PowerSource } from "./power-source";
+
+export class LightBulb {
+  protected readonly powerConsumption = 20;
+  constructor(private powerSource: PowerSource) {}
+
+  light() {
+    this.powerSource.consume(this.powerConsumption);
+  }
+
+  lightWithTimer(n: number) {
+    this.light();
+    if (n > 1) this.lightWithTimer(n - 1);
```

Comment:
To powinno działać z faktycznym upływem czasu.
Dodatkowo nie powinno zależeć od tego "na ile włączysz" żarówkę - w sensie na `n` sekund.
Ponieważ żarówkę wkręcasz - i jest.
A to że zapali się jedynie na 5 sekund - wynika z ilości dostępnej mocy w `powerSource`.

Reply:
Tutaj mój błąd - to że ma to świeci `n` sekund - wynikało z treści zadania - więc może tak zostać :)

Reply:
Jednak i tak - brakuje tutaj fizycznego upływu czasu.